app/src/main/java/com/example/concordia_campus_guide/Helper/IndoorPathHeuristic.java

```diff
@@ -12,23 +12,35 @@
 
     Context context;
 
-    public IndoorPathHeuristic(Context context){
+    public IndoorPathHeuristic(Context context) {
         this.context = context;
     }
```

Comment:
@thomasgauvin  has created used factory design pattern to to create the view models that use the database. @thomasgauvin  do you think we can use it here too to avoid using context ?

Reply:
Absolutely, this class can be rewritten using the viewmodelfactory because we only use the context to get the dao. So we could store the AppDatabase as an attribute to the IndoorPathHeuristic instead of the Context.

Reply:
Instead of using a factory to do so, I think passing the database directly as a parameter is a better solution for this case. I will update this PR with my changes for that!